### PR TITLE
chore(v2): Update webpack-related dependencies, pre-work to migrate to Webpack 5

### PR DESCRIPTION
**Pull request by RDIL**
## Motivation

Begin to implement #4027 by updating the webpack plugins & related code.

**Notes for webpack 5 upgrade**:
- mini-css-extract-plugin on latest BREAKS SEARCH stuff in the production build
- `html-webpack-plugin` will need an update
- `null-loader` should be removed
- `terser-webpack-plugin` on latest just doesn't work

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan
I've confirmed it still works just fine on the current webpack version (v4).

## Related PRs

n/a

### CODE REVIEW

File: packages/docusaurus/src/webpack/utils.ts [hunk 1]
```diff
@@ -46,13 +46,9 @@ export function getStyleLoaders(
     ];
   }
 
-  const isProd = process.env.NODE_ENV === 'production';
   const loaders = [
     {
       loader: MiniCssExtractPlugin.loader,
-      options: {
-        hmr: !isProd,
```
Comment (RDIL) [inline packages/docusaurus/src/webpack/utils.ts:54]: This is now automatically activated if the webpack dev server has it enabled.

File: packages/docusaurus/src/webpack/base.ts [hunk 2]
```diff
@@ -180,10 +180,10 @@ export function createBaseConfig(
               localIdentName: isProd
                 ? `[local]_[hash:base64:4]`
                 : `[local]_[path]`,
+              exportOnlyLocals: isServer,
             },
             importLoaders: 1,
             sourceMap: !isProd,
-            onlyLocals: isServer,
```
Comment (RDIL) [inline packages/docusaurus/src/webpack/base.ts:186]: This option was just moved and renamed.

File: packages/docusaurus-utils/package.json [hunk 3]
```diff
@@ -19,7 +19,7 @@
   "license": "MIT",
   "dependencies": {
     "@docusaurus/types": "2.0.0-alpha.70",
-    "chalk": "^3.0.0",
+    "chalk": "^4.1.0",
```
Comment (RDIL) [inline packages/docusaurus-utils/package.json:22]: Makes the version of chalk match across packages.

**Comment by semoal** [comment]
About `terser`: If you are using webpack v5 or above you do not need to install this plugin. Webpack v5 comes with the latest terser-webpack-plugin out of the box. Using Webpack v4

About `html-webpack-plugin`: actually you have to install @next version to make it compatible with webpack 5

**Comment by RDIL** [comment]
> About `terser`: If you are using webpack v5 or above you do not need to install this plugin. Webpack v5 comes with the latest terser-webpack-plugin out of the box. Using Webpack v4
> 
> About `html-webpack-plugin`: actually you have to install @next version to make it compatible with webpack 5

Yes, but this PR is just to bump up the non-critical dependencies to make the actual webpack 5 PR smaller.

**Comment by slorber** [comment]
Ah I see, that's why the deploy preview looks fine to me 😅

Thanks for your work, will review it soon

**Comment by slorber** [comment]
LGTM didn't see any issue

**Pull request merged**